wiring/src/spark_wiring_ble.cpp

```diff
@@ -562,6 +562,11 @@ BleAdvertisingData::BleAdvertisingData()
         : selfData_() {
     uint8_t flag = BLE_SIG_ADV_FLAGS_LE_ONLY_GENERAL_DISC_MODE;
     append(BleAdvertisingDataType::FLAGS, &flag, sizeof(uint8_t));
+#if HAL_PLATFORM_RTL872X
```

Comment:
Maybe let's introduce some new define in `ble_hal_impl.h` for this and set to either `BLE_MAX_ADV_DATA_LEN` or `BLE_MAX_ADV_DATA_LEN_EXT` depending on the platform?